application/src/main/java/org/togetherjava/tjbot/features/javamail/RSSHandlerRoutine.java

```diff
@@ -0,0 +1,311 @@
+                lastPostedDate = pubDate;
+            }
+        }
+
+        // Finally, save the last posted date to the database.
+        DateTimeFormatter dateTimeFormatter =
+                DateTimeFormatter.ofPattern(feedConfig.dateFormatterPattern());
+        String lastDateStr = lastPostedDate.format(dateTimeFormatter);
+        if (dateResult == null) {
+            database.write(context -> context.newRecord(RSS_FEED)
+                .setUrl(feedConfig.url())
+                .setLastDate(lastDateStr)
+                .insert());
+            return;
+        }
+
+        // If we already have an existing record with the given URL,
+        // now is the time to update it
+        database.write(context -> context.update(RSS_FEED)
+            .set(RSS_FEED.LAST_DATE, lastDateStr)
+            .where(RSS_FEED.URL.eq(feedConfig.url()))
+            .executeAsync());
+    }
+
+    /**
+     * Attempts to get a {@link ZonedDateTime} from an {@link Item} with a provided string date time
+     * format.
+     * <p>
+     * If either of the function inputs are null, the oldest-possible {@link ZonedDateTime} will get
+     * returned instead.
+     *
+     * @return The computed {@link ZonedDateTime}
+     */
+    private static ZonedDateTime getDateTimeFromItem(Item item, String dateTimeFormat) {
+        String pubDate = item.getPubDate().orElse(null);
+
+        if (pubDate == null || dateTimeFormat == null) {
+            return ZONED_TIME_MIN;
+        }
+
+        return getLocalDateTime(pubDate, dateTimeFormat);
+    }
+
+    /**
+     * Given a list of RSS feed items, this function checks if the dates are valid
+     * <p>
+     * This assumes that all items share the same date format (as is usual in an RSS feed response),
+     * therefore it only checks the first item's date for the final result.
+     */
+    private static boolean isValidDateFormat(List<Item> rssFeeds, RSSFeed feedConfig) {
+        try {
+            final Item firstRssFeed = rssFeeds.getFirst();
+            String firstRssFeedPubDate = firstRssFeed.getPubDate().orElse(null);
+
+            if (firstRssFeedPubDate == null) {
+                return false;
+            }
+
+            getLocalDateTime(firstRssFeedPubDate, feedConfig.dateFormatterPattern());
+        } catch (Exception e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Attempts to find a text channel from a given RSS feed configuration.
+     */
+    private Optional<TextChannel> getTextChannelFromFeed(JDA jda, RSSFeed feed) {
+        // Attempt to find the target channel
+        if (feed.targetChannelPattern() != null) {
+            return jda.getTextChannelCache()
+                .stream()
+                .filter(channel -> targetChannelPatterns.get(feed).test(channel.getName()))
+                .findFirst();
+        }
+
+        // If the target channel was not found, use the fallback
+        return jda.getTextChannelCache()
```

Comment:
Maybe put this second part in an else so the intent is clearer that this default part happens because the first part couldn't happen, and you might even be able to remove the comments.

Reply:
Done!